eatin-client/src/components/functions/SavedStateInRecipes.ts

```diff
@@ -0,0 +1,102 @@
+import { useCallback, useEffect, useState } from "react";
+import { useAuth } from "../../context/auth-context";
+import { useGetUserrecipesByUserIdQuery } from "../../generated/graphql";
+import { Recipe, RecipesSection } from "../types";
+
+export function useAddIsSavedToRecipesSection(givenRecipes: RecipesSection[]) {
+    const { currentUser } = useAuth();
+    const { data: userRecipesData } = useGetUserrecipesByUserIdQuery({
+        variables: { userID: currentUser ? currentUser.uid : "" },
+    });
+    const [updatedRecipes, setNewRecipes] = useState<RecipesSection[]>([]);
```

Comment:
in states the names should be the same: `[<name>, set<Name>]`
also, I would change the names so it would be more clear:
`updatedRecipes` --> `recipesWithIsSaved`